example/src/screens/APIClientRequestScreen.tsx

```diff
@@ -0,0 +1,161 @@
+    const { client, method } = route.params;
+    const [endpoint, setEndpoint] = useState("/api/v4/system/ping");
+    const [timeoutInterval, setTimeoutInterval] = useState(30);
+    const [body, setBody] = useState('{"login_id":"","password":""}');
+    const [requestHeaders, setRequestHeaders] = useState<Header[]>([]);
+    const [response, setResponse] = useState<ClientResponse>();
+    const [responseVisible, setResponseVisible] = useState(false);
+    const [retryPolicyConfiguration, setRetryPolicyConfiguration] = useState<
+        RetryPolicyConfiguration
+    >({
+        type: undefined,
+        retryLimit: 2,
+        exponentialBackoffBase: 2,
+        exponentialBackoffScale: 0.5,
+    });
+
+    const toggleRetryPolicyType = () =>
+        setRetryPolicyConfiguration({
+            ...retryPolicyConfiguration,
+            type: retryPolicyConfiguration.type
+                ? undefined
+                : Constants.EXPONENTIAL_RETRY,
+        });
+    const setRetryLimit = (retryLimit: number) =>
+        setRetryPolicyConfiguration({
+            ...retryPolicyConfiguration,
+            retryLimit,
+        });
+    const setExponentialBackoffBase = (exponentialBackoffBase: number) =>
+        setRetryPolicyConfiguration({
+            ...retryPolicyConfiguration,
+            exponentialBackoffBase,
+        });
+    const setExponentialBackoffScale = (exponentialBackoffScale: number) =>
+        setRetryPolicyConfiguration({
+            ...retryPolicyConfiguration,
+            exponentialBackoffScale,
+        });
+
+    const makeRequest = async () => {
+        const headers = parseHeaders(requestHeaders);
+        let options: RequestOptions = {
+            headers,
+            timeoutInterval,
+            retryPolicyConfiguration,
+        };
+        if (method !== METHODS.GET && body.length) {
+            try {
+                options.body = JSON.parse(body);
+            } catch (e) {
+                Alert.alert("Error parsing Body", e.message, [{ text: "OK" }], {
+                    cancelable: false,
+                });
+                return;
+            }
+        }
+
+        try {
+            let clientMethod;
+            switch (method) {
+                case METHODS.GET:
+                    clientMethod = client.get;
+                    break;
+                case METHODS.POST:
+                    clientMethod = client.post;
+                    break;
+                case METHODS.PUT:
+                    clientMethod = client.put;
+                    break;
+                case METHODS.PATCH:
+                    clientMethod = client.patch;
+                    break;
+                case METHODS.DELETE:
+                    clientMethod = client.delete;
+                    break;
+                default:
+                    throw new Error("Invalid request method");
+            }
+            var response = await clientMethod(endpoint, options);
```

Comment:
I'm wondering if it's worth investigating a way of calling the methods on `client` dynamically than having to do this workaround; being able to do `client.{method}(endpoint, options)` - just a note for the nice to have list.